handlers/node/node.go

```diff
@@ -133,31 +138,31 @@ func NodeEnrollRequest(respWriter http.ResponseWriter, request *http.Request) {
 			}
 
 			osc.SetTimestamp()
-			mu := sync.Mutex{}
 			// might be good to check for dupe hostnames here before ACTUALLY issuing new key
-			err := dyndb.UpsertClient(osc, dynSvc, &mu)
+			err := dyndb.UpsertClient(osc, dynSvc)
 			if err != nil {
 				nodeEnrollRequestLogger.WithFields(log.Fields{
 					"hostname": data.HostIdentifier,
 				}).Info("failed to upsert node")
 				return fmt.Errorf("node upsert failed: %s", err)
 			}
 
-			response.WriteCustomJSON(respWriter, EnrollRequestResponse{NodeKey: nodeKey, NodeInvalid: false})
+			//return invalid node response to client
+			response.WriteCustomJSON(respWriter, EnrollRequestResponse{NodeKey: nodeKey, NodeInvalid: nodeInvalid})
 		default:
 			nodeEnrollRequestLogger.WithFields(log.Fields{
 				"hostname": data.HostIdentifier,
 			}).Info("host already exists, setting host to existing node_key")
 			osc := ans[0]
 			osc.SetTimestamp()
-			mu := sync.Mutex{}
-			err := dyndb.UpsertClient(osc, dynSvc, &mu)
+			err := dyndb.UpsertClient(osc, dynSvc)
 			if err != nil {
 				nodeEnrollRequestLogger.Error(err)
 				return fmt.Errorf("node upsert failed: %s", err)
 			}
 			// might be good to check for dupe hostnames here before ACTUALLY issuing new key
-			response.WriteCustomJSON(respWriter, EnrollRequestResponse{NodeKey: ans[0].NodeKey, NodeInvalid: false})
+			//return a valid node response to client
+			response.WriteCustomJSON(respWriter, EnrollRequestResponse{NodeKey: ans[0].NodeKey, NodeInvalid: nodeValid})
```

Comment:
I know this is already merged, but this inverse logic is sorta confusing:

`NodeInvalid == nodeValid == false` (confusing because I wasn't expecting a var name `nodeValid` to equal `false` and `nodeInvalid` to equal `true`)

Would it be worth changing the `EnrollRequestResponse` struct to have a `NodeValid` var instead of `NodeInvalid` to make this more clear?

Reply:
@ryandeivert this actually is a product of the keys the osquery node is expecting.  Unfortunately the key is 
```json
"node_invalid": true
```
so it already starts with a negative.  It's entirely possible the best action here is to actually pass the correct boolean value, just for the sake of clarity, but I went back and forth with myself a few times and couldn't decide.  Totally open to a suggestion to change if you feel you've got something that works (and doesn't confuse the shit outa people like this does :) )

Reply:
ah damn you osquery! okay nevermind then :) I'll churn it over but this may be the best approach